packages/mgt-chat/src/components/ChatList/ChatList.tsx

```diff
@@ -187,16 +189,17 @@ export const ChatList = ({
       chatListClient.tearDown();
       setHeaderBannerMessage('We ran into a problem. Please close or refresh.');
     };
-  }, [chatListClient, onMessageReceived, onLoaded]);
+  }, [chatListClient, onMessageReceived, onLoaded, internalSelectedChatId]);
```

Comment:
Maybe what needs to happen is this useEffect needs to be broken up so that the `onStateChange` handlers can be updated without triggering the teardown any time they change. This also allows `onLoaded` and `onMessageReceived` callbacks to not use `useCallback` by the implementer in order to avoid teardowns.

Reply:
Maybe @seekdavidlee is handling this in his refactor

Reply:
We can also break up the useEffect, so that maybe there is one just for onMessageReceived which doesn't have anything to do with the teardown.

Reply:
@seekdavidlee 's PR is going to address the teardown